plugins/modules/idrac_diagnostics.py

```diff
@@ -0,0 +1,860 @@
+        job_uri = f"{res_uri[0]}/{OEM}/{MANUFACTURER}/{JOBS}{JOBS_EXPAND}"
+        job_resp = self.idrac.invoke_request(job_uri, "GET")
+        job_list = job_resp.json_data.get('Members', [])
+        job_id = ""
+        for jb in job_list:
+            if jb.get("JobType") == "RemoteDiagnostics" and jb.get("JobState") in ["Scheduled", "Running", "Starting", "New"]:
+                job_id = jb['Id']
+                job_dict = remove_key(jb, regex_pattern=ODATA_REGEX)
+                break
+        if self.module.check_mode and job_id:
+            self.module.exit_json(msg=ALREADY_RUN_MSG, job_details=job_dict, skipped=True)
+        if self.module.check_mode and not job_id:
+            self.module.exit_json(msg=CHANGES_FOUND_MSG, changed=True)
+        if job_id:
+            self.module.exit_json(msg=ALREADY_RUN_MSG, job_details=job_dict, skipped=True)
+
+
+class ExportDiagnostics(Diagnostics):
+
+    def execute(self):
+        self.test_network_share()
+        self.__get_export_diagnostics_url()
+        if self.module.check_mode:
+            self.perform_check_mode()
+        job_status = {}
+        share_type = self.module.params.get('share_parameters').get('share_type')
+        share_type_methods = {
+            "local": self.__export_diagnostics_local,
+            "http": self.__export_diagnostics_http,
+            "https": self.__export_diagnostics_http,
+            "cifs": self.__export_diagnostics_cifs,
+            "nfs": self.__export_diagnostics_nfs
+        }
+        export_diagnostics_status = share_type_methods[share_type]()
+        if share_type_methods[share_type] != self.__export_diagnostics_local:
+            job_status = self.get_job_status(export_diagnostics_status)
+        status = export_diagnostics_status.status_code
+        diagnostics_file_path = f"{self.share_name}/{self.file_name}"
+        if status in STATUS_SUCCESS:
+            msg = SUCCESS_EXPORT_MSG
+            job_details = job_status
+            return msg, job_details, diagnostics_file_path
+
+    def __export_diagnostics_local(self):
+        payload = {}
+        payload["ShareType"] = "Local"
+        file_path = self.module.params.get('share_parameters').get('share_name')
+        self.share_name = file_path.rstrip("/")
+        diagnostics_status = self.__export_diagnostics(payload)
+        diagnostics_file_name = payload.get("FileName")
+        diagnostics_data = self.idrac.invoke_request(diagnostics_status.headers.get("Location"), "GET")
+        file_name = os.path.join(file_path, diagnostics_file_name)
+        with open(file_name, "w") as fp:
+            fp.write(diagnostics_data.body.decode().replace("\r", ""))
+        return diagnostics_status
+
+    def __export_diagnostics_http(self):
+        payload = self.get_payload_details()
+        export_status = self.__export_diagnostics(payload)
+        share = self.module.params.get('share_parameters')
+        ip = config_ipv6(share.get('ip_address'))
```

Comment:
Good use of config_ipv6 utils :)

Reply:
btw thanks to @sachin-apa, he suggested me to use this.